src/magicbox/QueriesProcessor.ts

```diff
@@ -0,0 +1,91 @@
+import { defaults } from 'underscore';
+
+export interface IQueriesProcessorOptions {
+  timeout: number;
+}
+
+export enum QueryProcessResultStatus {
+  Finished = 'finished',
+  TimedOut = 'timedout',
+  Overriden = 'overriden'
```

Comment:
What does "overriden" mean in this context ?

Reply:
It means that the query was overriden by another call to `processQueries`. Would `Aborted` be better? This is also used when a process was manually overriden with `overrideIfProcessing`, would it be better to differentiate them?